app/controllers/admin/email_stats_controller.rb

```diff
@@ -0,0 +1,36 @@
+# ============================================================================
+# Project:   ShinyCMS (Ruby version)
+# File:      app/controllers/admin/email_stats_controller.rb
+# Purpose:   Controller for viewing email stats in ShinyCMS admin area
+#
+# Copyright: (c) 2009-2020 Denny de la Haye https://denny.me
+#
+# ShinyCMS is free software; you can redistribute it and/or
+# modify it under the terms of the GPL (version 2 or later).
+# ============================================================================
+class Admin::EmailStatsController < AdminController
+  before_action :set_ahoy_user
+
+  def index
+    page_num = params[ :page ] || 1
+    messages = Ahoy::Message
+    messages = messages.where( user: @ahoy_user ) if @ahoy_user
+    @messages = messages.order( 'sent_at desc' ).page( page_num )
+
+    authorise Ahoy::Message
```

Comment:
Although there are no db writes yet here, if there were, this delayed authorisation could allow unauthorised writes to the db.

Consider moving the `authorise Ahoy::Message` to the first line of this `index` action, to mitigate this and minimise the work done for an unauthorised user.